From 69a8cd81fb5354412e72002a4158169d9ea1d39a Mon Sep 17 00:00:00 2001
From: Honore Tricot <honorex.tricot@intel.com>
Date: Fri, 26 Oct 2012 09:29:02 +0200
Subject: [PORT FROM MAIN] wifi: hide password in logs

BZ: 64477

When the SET_NETWORK command is sent in order to set the password,
the password value appears in clear inside logs.

This fix catch the command SET_NETWORK + password attribute and remove
its value from string to send into log.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: yes

Change-Id: I15ebd0e5bfb45b211e9738ccbe908f3ddd23f605
Orig-Change-Id: Iad72ec90035180450ce3fdd4e2496357f63d7113
Signed-off-by: Paul Naoumenko <paulx.naoumenko@intel.com>
---
 wifi/wifi.c |   15 ++++++++++++++-
 1 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/wifi/wifi.c b/wifi/wifi.c
index 4fe4920..2e0242c 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -709,7 +709,20 @@ int wifi_connect_to_supplicant(const char *ifname)
 
 void log_cmd(const char *cmd)
 {
-    LOGI("CMD: %s\n", cmd);
+    if (strstr (cmd, "SET_NETWORK") && strstr(cmd, "password")) {
+        char *pbuf = malloc(strlen(cmd) + 1);
+        if (pbuf) {
+            strncpy(pbuf, cmd, strlen(cmd) + 1);
+            pbuf[strlen(cmd)]='\0';
+            char *p = strchr(pbuf, '\"');
+            if (p)
+                *p = '\0';
+            LOGI("CMD: %s\n", pbuf);
+        }
+        free(pbuf);
+    }
+    else
+        LOGI("CMD: %s\n", cmd);
 }
 
 void log_reply(char *reply, size_t *reply_len)
-- 
1.7.4.1

